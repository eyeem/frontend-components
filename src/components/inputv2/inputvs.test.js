import React from 'react';
import { render, getByLabelText, cleanup } from 'react-testing-library';

import Inputv2 from './inputv2';

afterEach(cleanup);

test('it renders default input', () => {
  const { container } = render(
    <Inputv2 name="name" inputProps={{ placeholder: 'Name' }} />
  );

  const label = getByLabelText(container, 'Name');
  const component = container.firstChild;

  expect(label).toBeTruthy();
  expect(component).toMatchSnapshot();
});

test('it accepts property disabled', () => {
  const { container } = render(
    <Inputv2 name="name" inputProps={{ placeholder: 'Name', disabled: true }} />
  );
  const input = container.querySelector('input');

  expect(input).toHaveProperty('disabled', true);
});
