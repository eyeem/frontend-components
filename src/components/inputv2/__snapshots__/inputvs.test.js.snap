// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders default input 1`] = `
.c2 {
  position: absolute;
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  top: 22px;
  -webkit-transform-origin: left;
  -ms-transform-origin: left;
  transform-origin: left;
  font-weight: initial;
  -webkit-transform: translateY(0) scale(1);
  -ms-transform: translateY(0) scale(1);
  transform: translateY(0) scale(1);
}

.c1 {
  padding: 12px 16px;
  position: relative;
  border-width: 1.5px;
  border-style: solid;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
}

.c3 {
  width: 100%;
  border: 0;
  line-height: 24px;
  font-size: 16px;
  padding-top: 5px;
  padding-bottom: 9px;
  padding-right: 36px;
  background-color: transparent !important;
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c3::-webkit-input-placeholder {
  color: transparent;
}

.c3::-moz-placeholder {
  color: transparent;
}

.c3:-ms-input-placeholder {
  color: transparent;
}

.c3::placeholder {
  color: transparent;
}

.c3:focus {
  outline: none;
}

.c0 {
  height: 96px;
  position: relative;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <label
      class="c2"
      for="name"
    >
      Name
    </label>
    <input
      class="c3"
      id="name"
      name="name"
      placeholder="Name"
      type="text"
      value=""
    />
  </div>
</div>
`;
